### PR TITLE
[wip] return status OK only when db connection is successful

### PR DESCRIPTION
**Pull request by batpad**
Refs https://github.com/osmlab/to-fix-backend/pull/113

Questions:

 - What status code should we return on failure?
 - @mcwhittemore could you help me out with the failing test? i.e. mocking the db connection not working and testing that the `/` status end-point correctly returns `NOT OK` (or whatever we want it to return) -- wasn't sure how to cleanly over-ride the db setup stuff in `lib/test.js`

cc @kepta

### CODE REVIEW

**Comment by mcwhittemore** [comment]
hmmm... yea... I'll take a stab at adding `assert.db.close()` or something like it.

**Comment by mcwhittemore** [comment]
@batpad - I took a stab at this today and I'm not really sure what to do... Sequelize doesn't seem to have a `reconnect` method.

Per what to return `next(err)` from the failed db request should do.

**Comment by mcwhittemore** [comment]
Lets ticket writing a failing test for this and come back to it later. This doesn't need to slow us down

File: lib/server.js [hunk 1]
```diff
@@ -22,8 +22,14 @@ server.use(bodyParser.json());
...
 
 server.get('/', function(req, res) {
-  // TODO: check that the database is up and running so the status will be valid
-  res.json({ status: 'OK' });
+  db
+    .authenticate()
+    .then(() => {
+      res.json({ status: 'OK' });
+    })
+    .catch(() => {
```
Comment (mcwhittemore) [inline lib/server.js]: passing the error along to next will work.

```
.catch(err => next(err));
```

**Review by mcwhittemore — APPROVED** [review]
After making the suggested change and cutting a ticket, this looks good.

**Comment by kepta** [comment]
Update: I sent a PR to semi unit test when `db.authenticate()` returns a rejected promise.
https://github.com/osmlab/to-fix-backend/pull/121

**Pull request merged**